lib/install/management/appliance.go

```diff
@@ -191,17 +191,30 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		}
 	}
 
+	//removes the vm from vsphere, but detaches the disks first
 	_, err = vm.WaitForResult(d.ctx, func(ctx context.Context) (tasks.Task, error) {
 		return vm.DeleteExceptDisks(ctx)
 	})
 	if err != nil {
-		err = errors.Errorf("Failed to destroy VM %q: %s", vm.Reference(), err)
-		err2 := vm.Unregister(d.ctx)
-		if err2 != nil {
-			return errors.Errorf("%s then failed to unregister VM: %s", err, err2)
+		f, ok := err.(types.HasFault)
```

Comment:
hmm.. for vic-machine, if we reach here, we already powered off the VM, and plan to delete the vm even there is error happens, I guess.
If user wants to delete the VCH together with container, if we differentiate the error type here, user might have to rerun the delete, and does not care about the error anyway.